aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py

```diff
@@ -0,0 +1,268 @@
+            for input_descriptor in input_descriptors:
+                for schema in input_descriptor.schemas:
+                    uri = schema.uri
+                    required = schema.required or True
+                    if required:
+                        # JSONLD Expanded URLs
+                        if "#" in uri:
+                            types.append(uri.split("#")[1])
+                        else:
+                            schema_ids.append(uri)
+            if len(types) == 0:
+                types = None
+            if len(schema_ids) == 0:
+                schema_ids = None
+            search = holder.search_credentials(
+                types=types,
+                schema_ids=schema_ids,
+            )
+            # Defaults to page_size but would like to include all
+            # For now, setting to 1000
+            max_results = 1000
+            records = await search.fetch(max_results)
+        except StorageNotFoundError as err:
+            raise V20PresFormatError(err)
+        # Selecting suite from claim_format
+        claim_format = pres_definition.fmt
+        proof_type = None
+        if claim_format:
+            if claim_format.ldp_vp:
+                for proof_req in claim_format.ldp_vp.get("proof_type"):
+                    if proof_req == "BbsBlsSignature2020":
+                        proof_type = "BbsBlsSignature2020"
+                        break
+                    elif proof_req == "Ed25519Signature2018":
+                        proof_type = "Ed25519Signature2018"
+                        break
+
+        dif_handler = DIFPresExchHandler(
+            self._profile, pres_signing_did=issuer_id, proof_type=proof_type
+        )
+
+        pres = await dif_handler.create_vp(
+            challenge=challenge,
+            domain=domain,
+            pd=pres_definition,
+            credentials=records,
+        )
+        return self.get_format_data(PRES_20, pres)
+
+    async def receive_pres(
+        self, message: V20Pres, pres_ex_record: V20PresExRecord
+    ) -> None:
+        """Receive a presentation, from message in context on manager creation."""
+
+    async def verify_pres(self, pres_ex_record: V20PresExRecord) -> V20PresExRecord:
+        """
+        Verify a presentation.
+
+        Args:
+            pres_ex_record: presentation exchange record
+                with presentation request and presentation to verify
+
+        Returns:
+            presentation exchange record, updated
+
+        """
+        async with self._profile.session() as session:
+            wallet = session.inject(BaseWallet)
+            dif_proof = V20Pres.deserialize(pres_ex_record.pres).attachment(self.format)
+            # challenge
+            challenge = None
+            req_pres_attach = pres_ex_record.pres_request.get(
+                "request_presentations~attach"
+            )
+            for pres_req in req_pres_attach:
+                if pres_req.get("@id") == "dif":
```

Comment:
Then this also wouldn't be needed. It is not required for the attachment to have an `@id` of `dif` so will probably lead to issues with any non-ACA-Py agent. If we use `.attachment(self.format)` it should return the correct attachment